### PR TITLE
[MLIR][GPUCommon] Remove typed pointer support

### PR DESCRIPTION
**Pull request by Dinistro**
This commit removes the GPUCommon's lowering support for typed pointers. Typed pointers have been deprecated for a while now and it's planned to soon remove them from the LLVM dialect.

Related PSA: https://discourse.llvm.org/t/psa-removal-of-typed-pointers-from-the-llvm-dialect/74502

### CODE REVIEW

**Comment by zero9178** [comment]
I see a few uses of `LLVMTypeConverter::getPointerType` in these files. E.g. here:
https://github.com/llvm/llvm-project/blob/fce83da0a429653c1b58a43554091ca037096c1c/mlir/lib/Conversion/GPUCommon/GPUToLLVMConversion.cpp#L85-L88
Can we also remove these in favour of plain and canonical `LLVM::LLVMPointerType::get` calls? I think this will also lead to a lot of cleanups in a few places (here and other dialects) as the element type is now dead code.

**Comment by aartbik** [comment]
Changes look good to me. Adding Guray to give the green light on this.

**Review by zero9178 — APPROVED** [review]
LGTM

File: mlir/include/mlir/Dialect/LLVMIR/LLVMDialect.h [hunk 1]
```diff
@@ -210,7 +210,7 @@ class GEPIndicesAdaptor {
 /// string (operations inserted at the builder insertion point).
 Value createGlobalString(Location loc, OpBuilder &builder, StringRef name,
                          StringRef value, Linkage linkage,
-                         bool useOpaquePointers);
+                         bool useOpaquePointers = true);
```
Comment (grypp) [inline mlir/include/mlir/Dialect/LLVMIR/LLVMDialect.h:213]: I am guessing this will be removed incrementally.

**Review by grypp — APPROVED** [review]
Thanks for doing this, it looks good.
Let me know if there's anything I can help with deprecating for `NVGPU` and `GPU` dialects.

File: mlir/include/mlir/Dialect/LLVMIR/LLVMDialect.h (hunk 1 above)
Comment (Dinistro) [reply mlir/include/mlir/Dialect/LLVMIR/LLVMDialect.h:213]: That's the plan, yes. We will be working on the LLVM dialect side once all users of typed pointers have been dropped.

**Comment by Dinistro** [comment]
> Thanks for doing this, it looks good. Let me know if there's anything I can help with deprecating for `NVGPU` and `GPU` dialects.

Thanks 😄 
I planned to do look into the other GPU related dialects this evening, if something comes up, I'll let you know.

**Pull request merged**